library/azure_rm_resource_facts.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_resource_facts
+version_added: "2.6"
+short_description: Generic facts of Azure resources.
+description:
+  - Obtain facts of any resource using Azure REST API.
+
+options:
+  url:
+    description:
+      - Azure RM Resource URL.
+  provider:
+    description:
```

Comment:
and more prorperties, like required or not, choices, default in all options

Reply:
default is none for all options